server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -271,6 +271,8 @@ static Setting<Integer> buildNumberOfShardsSetting() {
         Property.IndexScope
     );
 
+    public static final String SETTING_KNN_ENABLED = "index.knn";
```

Comment:
I am not sure if we should expose a plugin setting within core. Can you try adding in support for this behavior within the plugin using `IndexEventListener` APIs?

https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/shard/IndexEventListener.java